File: Examples/test-suite/java/nested_inheritance_interface_runme.java
import nested_inheritance_interface.*;
import java.util.Arrays;

public class nested_inheritance_interface_runme {

  static {
    try {
      System.loadLibrary("nested_inheritance_interface");
    } catch (UnsatisfiedLinkError e) {
      System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e);
      System.exit(1);
    }
  }

  private static void takeIA(IASwigInterface ia) {
  }

  public static void main(String argv[]) {
	Class[] BNInterfaces = B.N.class.getInterfaces();
    String expectedInterfacesString = "[interface nested_inheritance_interface.IASwigInterface]";
    String actualInterfacesString = Arrays.toString(BNInterfaces);
    if (!expectedInterfacesString.equals(actualInterfacesString))
      throw new RuntimeException("Expected interfaces for " + B.N.class.getName() + ": \n" + expectedInterfacesString + "\n" + "Actual interfaces: \n" + actualInterfacesString);

    if (!IASwigInterface.class.isInterface())
      throw new RuntimeException(IASwigInterface.class.getName() + " should be an interface but is not");

    // overloaded methods check
    B.N d = new B.N();
    takeIA(d);
  }
}

File: Examples/test-suite/kotlin/nested_inheritance_interface_runme.kt
@file:JvmName("nested_inheritance_interface_runme")

import nested_inheritance_interface.*

// The %interface(IA) feature is only applied for SWIGJAVA/SWIGCSHARP. For the Kotlin
// module %feature("flatnested") is applied instead and IA is generated as a plain base
// class (no IASwigInterface interface type, and N is flattened to a top-level class).
// The Java runme's interface-reflection assertions and B.N nesting are therefore not
// applicable here; this port simply exercises the generated class hierarchy (N : IA).

private fun takeIA(ia: IA) {
}

fun main() {
    try {
        System.loadLibrary("nested_inheritance_interface")
    } catch (e: UnsatisfiedLinkError) {
        System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e)
        kotlin.system.exitProcess(1)
    }

    val d = N()
    if (d !is IA)
        throw RuntimeException("N should be an IA")
    takeIA(d)
}
